Scottish-Tech-Army/Volunteer-app
# SVA-346 | Set up Expo

- [Description](#description)
  - [Type of change](#type-of-change)
- [Testing locally](#testing-locally)
- [Checklist](#checklist)

# Description

- Switches the app to using the Expo CLI and Expo Go with React Native -- you no longer use a local emulator to develop the app (don't try to run your normal emulator when reviewing this PR 🙂 )
- Should enable much easier development and deployment (this ticket/PR doesn't cover deployment -- there are separate tickets for App Store / Play Store deployment where we can address that)
- NB the front-end app test was failing because our Jest setup needs updating -- but as that test doesn't actually do anything useful yet, and fixing Jest feels like it's outside the scope of this PR, disabled the front-end app test script for now until we're ready to re-enable it (when we properly set up front-end app tests)
- Also deleted a few other unused files e.g. translations, dummy user data

Fixes # SVA-346

## Type of change


# Testing locally

I've tested this on Linux with an iPhone.  It would be great if at least one reviewer can try it on Windows and at least one reviewer with an Android phone, to make sure both of these work.

1. Add an `API_TUNNEL_SUBDOMAIN` variable to your `api/.env` file.  For this value, you should enter your own name in lowercase with only dashes in between, followed by a random string of letters and numbers. For example if your name is Nadia Bloggs: `API_TUNNEL_SUBDOMAIN="nadia-bloggs-dfkjhdfkj3847594385ksjksd"`
2. Remove any comments in your `api/.env` file (comments start with the `#` character)
3. In the `api` directory do `npm install`
4. In the `api` directory do `npm start`.  Leave this running.
5. In another terminal window in the `api` directory run the command `npm run tunnel-linux` (if you are on a Mac or Linux) or `npm run tunnel-windows` (if you are on Windows).  Make a note of the URL it tells you.
6. Update your `app/src/Config/index.ts` file -- change the `STA_BASE_URL` value to the URL from the previous step (e.g. `https://nadia-bloggs-dfkjhdfkj3847594385ksjksd.loca.lt` -- note this should be https and should **not** include a port number).  Leave this running.  *(NB you might want to keep note of the existing URL you were using, in case you need this for development on other branches not yet using Expo, until this PR has been merged)*
7. [Install the Expo Go app on your iOS or Android phone](https://expo.dev/client) -- if you don't have an Expo account already, you'll also need to [create one](https://expo.dev/signup)
8. In a new terminal window, in the `app` directory do `npm install` *NB I've seen some warnings about conflicting peer dependencies, to do with some React packages used by NativeBase, but not errors*
9. Then in the `app` directory do `npm start`
10. Connect your phone:
- iPhone: open the camera and scan the QR code, tap on the link and it should open up the app in Expo Go
- Android: open the Expo Go app itself and you can scan the QR code
You should now see your local development version of the app on your phone.  Please check that the app's working as normal.  See tips in the updated APP_DEVELOPMENT.md file in this PR if Expo Go gets disconnected or isn't showing changes.

# Checklist

- [] I have addressed accessibility, if needed
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
